src/inmanta/agent/handler.py

```diff
@@ -159,7 +160,46 @@ def bound(**kwds):
 
 
 @stable_api
-class HandlerContext(object):
+class IBasicLogging(ABC):
+    """
+    Minimal logging interface exposing logging methods for commonly used
+    logging levels.
+
+    Child classes are responsible for implementing a _log_msg method with the
+    concrete logging implementation.
+    """
+
+    def critical(self, msg: str, *args: object, **kwargs: object) -> None:
+        self._log_msg(logging.CRITICAL, msg, args, kwargs)
+
+    def error(self, msg: str, *args: object, **kwargs: object) -> None:
+        self._log_msg(logging.ERROR, msg, args, kwargs)
+
+    def warning(self, msg: str, *args: object, **kwargs: object) -> None:
+        self._log_msg(logging.WARNING, msg, args, kwargs)
+
+    def info(self, msg: str, *args: object, **kwargs: object) -> None:
+        self._log_msg(logging.INFO, msg, args, kwargs)
+
+    def debug(self, msg: str, *args: object, **kwargs: object) -> None:
+        self._log_msg(logging.DEBUG, msg, args, kwargs)
+
+    def exception(self, msg: str, *args: object, exc_info: bool = True, **kwargs: object) -> None:
+        self.error(msg, *args, exc_info=exc_info, **kwargs)
+
+    @abstractmethod
+    def _log_msg(
+        self,
+        level: int,
+        msg: str,
+        *args,
+        **kwargs,
+    ) -> None:
+        raise NotImplementedError
+
+
+@stable_api
+class HandlerContext(IBasicLogging, object):
```

Comment:
```suggestion
class HandlerContext(IBasicLogging):
```
Let's clean this up while we're at it. I think this was required in Python 3.5 or something so it hasn't been for a long time.